### PR TITLE
Correct description of how many Delayed Retries are performed

### PR DESCRIPTION
**Pull request by matthewhermann**
The [documentation for the Default Recoverability Policy](https://docs.particular.net/nservicebus/recoverability/custom-recoverability-policy?) is correct when it describes Immediate Retries:

> If the ImmediateProcessingFailures value is less or equal to the configured MaxNumberOfRetries for Immediate Retries, then the ImmediateRetry action is returned.

[Relevant code](https://github.com/Particular/NServiceBus/blob/c936c938c76f6a2e2e98a2849a9f029bb530e09b/src/NServiceBus.Core/Recoverability/DefaultRecoverabilityPolicy.cs#L30-L33):

```c#
if (errorContext.ImmediateProcessingFailures <= config.Immediate.MaxNumberOfRetries)
{
    return RecoverabilityAction.ImmediateRetry();
}
```

But it incorrectly uses "less or equal" when describing Delayed Retires:

> If the DelayedDeliveriesPerformed is less or equal than MaxNumberOfRetries and the message hasn't reached the maximum time allowed for retries (24 hours), then the DelayedRetry action is returned.

[The code](https://github.com/Particular/NServiceBus/blob/c936c938c76f6a2e2e98a2849a9f029bb530e09b/src/NServiceBus.Core/Recoverability/DefaultRecoverabilityPolicy.cs#L52-L55) actually stops when DelayedDeliveriesPerformed equals MaxNumberOfRetries

```c#
if (delayedDeliveriesPerformed >= config.MaxNumberOfRetries)
{
    return false;
}
```

The snippet for the for the fully customised Recoverability Policy also uses this "less or equal" - this isn't necessarily incorrect, as the snippet is _not_ trying to show the Default Recoverability Policy, but it can lead to confusion.

### CODE REVIEW

File: nservicebus/recoverability/index_retrycount_core_[6.0,).partial.md
Comment (adamralph) [inline nservicebus/recoverability/index_retrycount_core_[6.0,).partial.md:23]: What is the reason for switching these values?

**Review by adamralph — COMMENTED** [review]
Thanks very much for raising this @matthewhermann.

Your justification for this change makes sense to me. I'd just like to get another pair of eyes on it to confirm before merging.

File: nservicebus/recoverability/index_retrycount_core_[6.0,).partial.md
Comment (matthewhermann) [reply nservicebus/recoverability/index_retrycount_core_[6.0,).partial.md:23]: It looks like these were just in the wrong columns - ImmediateRetries is the first column, and DelayedRetries is the second column.

Elsewhere in the page it [mentions the default for Immediate Retries is 5](https://docs.particular.net/nservicebus/recoverability/#immediate-retries):

> By default up to five immediate retries are performed if the message processing results in exception being thrown.

Implies the [default for Delayed Retries is 3](https://docs.particular.net/nservicebus/recoverability/#delayed-retries):

> Delayed Retries introduces another level of retry mechanism for messages that fail processing. Delayed Retries schedules message delivery to the endpoint's input queue with increasing delay, by default first with 10 seconds delay, then 20, and lastly with 30 seconds delay.

And then [states 5 for immediate and 3 for delayed](https://docs.particular.net/nservicebus/recoverability/#total-number-of-possible-retries-scale-out-multiplier):

> When taking the default values for immediate and delayed retries (five and three, respectively) and 6 instances the total number of attempts will be a minumum of (5+1)(3+1)=24 attempts and a maximum of 246=144 attempts.

Comment (adamralph) [reply nservicebus/recoverability/index_retrycount_core_[6.0,).partial.md:23]: 👍 thanks for the explanation @matthewhermann, that makes sense.

Comment (adamralph) [reply nservicebus/recoverability/index_retrycount_core_[6.0,).partial.md:23]: Incidentally, the `*` characters in that last sentence are being interpreted as italics markdown. I've fixed that in https://github.com/Particular/docs.particular.net/pull/5013

Comment (matthewhermann) [reply nservicebus/recoverability/index_retrycount_core_[6.0,).partial.md:23]: Oh yes, I missed that - nice one 👍

**Review by tmasternak — APPROVED** [review]

**Comment by tmasternak** [comment]
@matthewhermann thank you very much for the fixes!

**Pull request merged**
